src/Microsoft.Health.Fhir.Core/Features/Search/Converters/MoneyNodeToQuantitySearchValueTypeConverter.cs

```diff
@@ -0,0 +1,61 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.FhirPath;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Converters
+{
+    /// <summary>
+    /// A converter used to convert from <see cref="Money"/> to a list of <see cref="QuantitySearchValue"/>.
+    /// </summary>
+    public class MoneyNodeToQuantitySearchValueTypeConverter : FhirNodeToSearchValueTypeConverter<QuantitySearchValue>
+    {
+        public MoneyNodeToQuantitySearchValueTypeConverter()
+            : base("Money")
+        {
+        }
+
+        protected override IEnumerable<ISearchValue> Convert(ITypedElement value)
+        {
+            var decimalValue = (decimal?)value.Scalar("value");
+
+            if (decimalValue == null)
+            {
+                yield break;
+            }
+
+            var currency = value.Scalar("currency")?.ToString();
+
+            // Currency information is specified differently if we are running STU3.
```

Comment:
Currently looking into adding an `#if Stu3` conditional compile constant here - I think it might add a bit of clarity.

Reply:
You can't add it in Core, it's only available in Shared (these projects compile into FHIR specific assemblies). In core you can use ModelInfoProvider.Version

Reply:
Ah, thank you. Added in 0ed9dcc